### DISCUSSION TITLE
Use Struct inside a Struct

### DISCUSSION THREAD

**Discussion by fosgate29** [Consensys/gnark#122] [Q&A] [answered]
HI. I am trying to building a circuit that can have a variable number of inputs. For example:
1) It has a fixed number of PublicKey variable. In this case, 3.
```go
type PublicKey = eddsa.PublicKey
type exoticCircuit struct {
	quoteQuery            frontend.Variable `gnark:",public"`  
	PublicKeyCpt1       PublicKey            `gnark:",public"`
	PublicKeyCpt2       PublicKey            `gnark:",public"`
	PublicKeyCpt3       PublicKey            `gnark:",public"`
}
```
My dounbt is: Can I have an array instead of having fixed 3 variables, PublicKeyCpt1, PublicKeyCpt2 and PublicKeyCpt3, I would have only one: **PublicKeyCpt       PublicKey            `gnark:",public"`**

I tried this code below, but didn't work:
```go
type PublicKey = eddsa.PublicKey
type PublicKeyCptArray struct {
    PublicKeyCpt  []PublicKey
}
type exoticCircuit struct {
	quoteQuery            frontend.Variable `gnark:",public"`  
	publicKeyCpt         PublicKeyCptArray      `gnark:",public"`
}
```
I tried but it is saying **invalid operation: cannot index circuit.publicKeyCpt (variable of type PublicKeyCptArray)**

Sorry, but I am new to Go and I am not sure if it is how gnark is suppose to work or if I am coding it wrong.

thanks for the help
LUiz

**Comment by ThomasPiellard** [comment, marked as answer]
Hi @fosgate29 ,

in your case since you need only 3 PublicKeys the simplest solution is to write it like this:
```
type PublicKey = eddsa.PublicKey

type exoticCircuit struct {
	QuoteQuery   frontend.Variable `gnark:",public"`
	PublicKeyCpt [3]PublicKey      `gnark:",public"`
}
```

Also the fields in the struct `exoticCircuit` must be exported for the circuit to be compiled, to do that make sure the name of the fields start with an upper case.

 Let us know if you still have issues.

**Reply by fosgate29** [reply]
Thanks. I was of great help.
